src/lib/y2network/sysconfig/dns_reader.rb

```diff
@@ -126,10 +127,15 @@ def searchlist
 
       # Returns whether the hostname should be taken from DHCP
       #
-      # @return [Boolean]
+      # @return [String,:any,:none] Interface to set the hostname based on DHCP settings;
+      #   :any for any interface; :none for ignoring the hostname assigned via DHCP
       def dhcp_hostname
         value = Yast::SCR.Read(Yast::Path.new(".sysconfig.network.dhcp.DHCLIENT_SET_HOSTNAME"))
-        value == "yes"
+        return :any if value == "yes"
+        files = InterfaceFile.all
+        files.each(&:load)
+        file = files.find { |f| f.dhclient_set_hostname == "yes" }
```

Comment:
NP: more effective will be to have load in find, so if there are 100 files and first contain this entry, it will be loaded faster.